### PR TITLE
cnct: remove nursery dependency from success resolver

### PR DESCRIPTION
**Pull request by joostjager**

### CODE REVIEW

File: contractcourt/htlc_success_resolver.go [hunk 1]
```diff
@@ -99,159 +93,108 @@ func (h *htlcSuccessResolver) Resolve() (ContractResolver, error) {
...
-		)
-		if err != nil {
+	// Wait for the sweep result.
+	select {
+	case result, ok := <-resultChan:
+		if !ok {
+			return nil, errResolverShuttingDown
+		}
+		if result.Err != nil {
 			return nil, err
```
Comment (chokoboko) [inline contractcourt/htlc_success_resolver.go]: Shouldn't this return result.Err?

Comment (joostjager) [reply contractcourt/htlc_success_resolver.go]: Nice catch, thank you. Fixed.

**Comment by joostjager** [comment]
Todo: figure out migration path when nursery is already trying to sweep

**Comment by joostjager** [comment]
For migration, it is probably enough to just run https://github.com/lightningnetwork/lnd/blob/2cd26d75564345c0c37fa83785ebe8e4f1fb8f28/sweep/store.go#L100 again so that sweeper recognizes nursery transactions as its own.

Other than that, the (new) resolvers are stateless and will pick up any state the resolution process was left in.

**Comment by joostjager** [comment]
Code got outdated because we didn't prioritize this. Closing PR now, because priority is still low. If that changes in the future, it can still be found as a reference on the parent issue #3688.

**Pull request closed without merging**